test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -213,6 +213,11 @@ object Assertion {
   final def containsString(element: String): Assertion[String] =
     Assertion.assertion("containsString")(param(element))(_.contains(element))
 
+  final def containsTheSameElements[A](other: Iterable[A]): Assertion[Iterable[A]] =
```

Comment:
I think we need to do a little more here because we want to make sure that in addition to having all the same elements we have the same number of each element. For example, right now `List(1, 1, 2, 3)` would be counted as containing the same elements as `List(1, 2, 3, 3)`.

Reply:
The latest change will make the assertion fail for this case.

Reply:
What do you think about calling this just `sameElements`? It is shorter and what @ghostdogpr originally suggested. Downside is it is the same as the collection method `sameElements` which checks the elements in order.

Reply:
My understanding is that assertion names should be verbs, that's why I used this. Eg: in a previous PR I had `nonEmpty` and we changed it to `isNonEmpty`.   Maybe rename it to `hasSameElements`?

Reply:
Yeah, that is definitely true though we have some exceptions like `forall`. I'm fine with `hasSameElements` unless anyone else has other ideas.

Reply:
+1 for `hasSameElements`

Reply:
Alright. Well that is settled. 😃